### PR TITLE
Ignore paths that cannot be stat-ed

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

`conda clean` produced an `UnboundLocalError` if any errors occurred while stat-ing paths to remove. This change captures this error for display but otherwise ignores the issue and lets the user resolve the issue on their own.

Resolves https://github.com/conda/conda/issues/12535

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: conda/cli/main_clean.py [hunk 1]
```diff
@@ -128,8 +130,8 @@ def rm_pkgs(
     from ..utils import human_bytes
 
     if verbose and warnings:
-        for fn, exception in warnings:
-            print(exception)
+        for path, exception in warnings:
+            print(path, exception)
```
Comment (dholth) [inline conda/cli/main_clean.py:134]: Call this function with verbose and warnings in the tests

File: conda/cli/main_clean.py [hunk 2]
```diff
@@ -1,41 +1,43 @@
...
-from typing import Any, Dict, Iterable, List, Tuple
+from typing import Any, Iterable
 import sys
 
 from ..base.constants import CONDA_PACKAGE_EXTENSIONS, CONDA_TEMP_EXTENSIONS, CONDA_LOGS_DIR
 from ..base.context import context
 
 log = getLogger(__name__)
-_EXTS = (*CONDA_PACKAGE_EXTENSIONS, *(f"{e}.part" for e in CONDA_PACKAGE_EXTENSIONS))
+_EXTENSIONS = (*CONDA_PACKAGE_EXTENSIONS, *(f"{ext}.part" for ext in CONDA_PACKAGE_EXTENSIONS))
```
Comment (dholth) [inline conda/cli/main_clean.py:15]: Do we use `.part` while downloading packages? It might be a good idea but I don't recall the code that does so.

Comment (kenodegard) [reply conda/cli/main_clean.py]: not sure

File: conda/cli/main_clean.py [hunk 3]
```diff
@@ -132,8 +137,8 @@ def rm_pkgs(
     from .common import confirm_yn
 
     if verbose and warnings:
-        for fn, exception in warnings:
-            print(exception)
+        for warning in warnings:
+            print(warning)
```
Comment (travishathaway) [inline conda/cli/main_clean.py:141]: Is there relatively straightforward way to add coverage to these lines?

**Review by jezdez — APPROVED** [review]

**Pull request merged**